dlab_core/infrastructure/services.py

```diff
@@ -0,0 +1,182 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# ******************************************************************************
+import requests
+import urllib3
+
+from dlab_core.domain.exceptions import DLabException
+from dlab_core.domain.logger import INFO, ERROR
+from dlab_core.infrastructure.logger import StreamLogBuilder
+
+from jose import jwt
+from jose.exceptions import (
+    JOSEError, JWTError, ExpiredSignatureError, JWTClaimsError
+)
+
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+class KeyCloakError(DLabException):
+    def __init__(self, *args, **kwargs):
+        self.message = self.message.format(*args, **kwargs)
+
+    def __str__(self):
+        return self.message
+
+
+class InvalidParameterFormatError(KeyCloakError):
+    message = 'Parameter {value} is not of {expected_type} type: {actual_type}'
+
+
+class ConnectionError(KeyCloakError):
+    message = 'Server responded with {status_code} - {content}'
+
+
+def check_response(func):
+    """Wrapper that checks if response returned by wrapped function is good
+
+    :type func: function
+    :param func: function to wrap
+
+    :rtype: requests.Response
+    :return: :class:`Response <Response>` object
+
+    :raises InvalidResponseError: if response status code is not 200
+    """
+    def wrapper(*args, **kwargs):
+        response = func(*args, **kwargs)
+        if not response.ok:
+            raise ConnectionError(
+                status_code=response.status_code,
+                content=response.content.decode('utf8')
+            )
+        return response
+
+    return wrapper
+
+
+class KeyCloak(object):
+    """Client for KeyCloak interaction
+
+    :type keycloak_host: str
+    :param keycloak_host: IP or hostname for KeyCloak server to use (e.g.
+`https://12.12.12.12/auth/`)
+
+    :type realm_name: str
+    :param realm_name: name of the KeyCloak Realm to use
+
+    :type client_id: str
+    :param client_id: KeyCloak Client to use for token introspection
+
+    :type client_secret: str
+    :param client_secret: KeyCloak Client secret string
+    """
+    def __init__(self, keycloak_host, realm_name, client_id, client_secret):
+        self.check_args_types(locals(), str)
+        self._realm_address = '{}realms/{}'.format(keycloak_host, realm_name)
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self.logger = StreamLogBuilder('keycloak', INFO).logging
```

Comment:
you do not need to create a logger
logger needs to be in args
in your case, logger will be different as a global logger and we will not have the possibility to control it behaviour

Reply:
Should I remove logging or pass logger as a constructor parameter?

Reply:
I removed logger until global logger is implemented